static/app/components/profiling/flamegraph.tsx

```diff
@@ -28,18 +28,24 @@ function Flamegraph(props: FlamegraphProps): ReactElement {
   const [activeProfileIndex, setActiveProfileIndex] = useState<number | null>(null);
   const [importedProfiles, setImportedProfiles] = useState<ProfileGroup | null>(null);
 
-  const flamegraph = useMemo(() => {
+  const profiles = useMemo(
     // once an import occurs, it will always take precedence over the profile in the props
-    const profiles = importedProfiles ?? props.profiles;
+    () => importedProfiles ?? props.profiles,
+    [importedProfiles, props.profiles]
+  );
 
+  const profileIndex = useMemo(
     // if the activeProfileIndex is null, use the activeProfileIndex from the profile group
-    const profileIndex = activeProfileIndex ?? profiles.activeProfileIndex;
+    () => activeProfileIndex ?? profiles.activeProfileIndex,
```

Comment:
We already store this on the FlamegraphModel. So below you could just flamegraph.profileIndex instead of storing this separately so that we keep the flamegraph as the source of truth